File: examples/quickhr/backend/src/data/employees.ts
/**
 * Mock employee data for QuickHR demo
 */

export interface Employee {
  id: string;
  name: string;
  email: string;
  role: string;
  department: string;
  startDate: string;
  managerId: string | null;
  status: "active" | "inactive" | "onboarding";
}

export const employees: Employee[] = [
  {
    id: "EMP001",
    name: "Marco Rossi",
    email: "marco.rossi@company.com",
    role: "Software Engineer",
    department: "Engineering",
    startDate: "2025-02-15",
    managerId: "EMP003",
    status: "onboarding",
  },
  {
    id: "EMP002",
    name: "Giulia Bianchi",
    email: "giulia.bianchi@company.com",
    role: "Product Manager",
    department: "Product",
    startDate: "2024-03-01",
    managerId: "EMP005",
    status: "active",
  },
  {
    id: "EMP003",
    name: "Alessandro Romano",
    email: "alessandro.romano@company.com",
    role: "Engineering Manager",
    department: "Engineering",
    startDate: "2023-01-15",
    managerId: "EMP005",
    status: "active",
  },
  {
    id: "EMP004",
    name: "Carlos Alcaraz",
    email: "carlos.alcaraz@company.com",
    role: "UX Designer",
    department: "Design",
    startDate: "2024-06-01",
    managerId: "EMP002",
    status: "active",
  },
  {
    id: "EMP005",
    name: "Luca Ferrari",
    email: "luca.ferrari@company.com",
    role: "CTO",
    department: "Executive",
    startDate: "2022-01-01",
    managerId: null,
    status: "active",
  },
  {
    id: "EMP006",
    name: "Roberto Stagi",
    email: "roberto.stagi@company.com",
    role: "HR Manager",
    department: "HR",
    startDate: "2023-06-01",
    managerId: "EMP005",
    status: "active",
  },
  {
    id: "EMP007",
    name: "Jannik Sinner",
    email: "jannik.sinner@company.com",
    role: "Sales Rep",
    department: "Sales",
    startDate: "2024-01-15",
    managerId: "EMP008",
    status: "active",
  },
  {
    id: "EMP008",
    name: "Francesca Gallo",
    email: "francesca.gallo@company.com",
    role: "Sales Director",
    department: "Sales",
    startDate: "2023-03-01",
    managerId: "EMP005",
    status: "active",
  },
  {
    id: "EMP009",
    name: "Andrea Conti",
    email: "andrea.conti@company.com",
    role: "DevOps Engineer",
    department: "Engineering",
    startDate: "2024-09-01",
    managerId: "EMP003",
    status: "active",
  },
  {
    id: "EMP010",
    name: "Laura Russo",
    email: "laura.russo@company.com",
    role: "Accountant",
    department: "Finance",
    startDate: "2024-04-01",
    managerId: "EMP005",
    status: "active",
  },
];

// Helper functions for data manipulation
export function getEmployeeById(id: string): Employee | undefined {
  return employees.find((e) => e.id === id);
}

export function getEmployeesByDepartment(department: string): Employee[] {
  return employees.filter((e) => e.department === department);
}

export function getEmployeesByStatus(
  status: Employee["status"]
): Employee[] {
  return employees.filter((e) => e.status === status);
}

export function addEmployee(employee: Omit<Employee, "id">): Employee {
  const newId = `EMP${String(employees.length + 1).padStart(3, "0")}`;
  const newEmployee = { ...employee, id: newId };
  employees.push(newEmployee);
  return newEmployee;
}

export function updateEmployee(
  id: string,
  updates: Partial<Employee>
): Employee | undefined {
  const index = employees.findIndex((e) => e.id === id);
  if (index === -1) return undefined;
  employees[index] = { ...employees[index], ...updates };
  return employees[index];
}

File: examples/py-langgraph/src/quickhr/data/employees.py
from __future__ import annotations

from typing import TypedDict


class Employee(TypedDict):
    id: str
    name: str
    email: str
    role: str
    department: str
    start_date: str
    manager_id: str | None
    status: str  # "active" | "inactive" | "onboarding"


employees: list[Employee] = [
    {
        "id": "EMP001",
        "name": "Marco Rossi",
        "email": "marco.rossi@company.com",
        "role": "Software Engineer",
        "department": "Engineering",
        "start_date": "2025-02-15",
        "manager_id": "EMP003",
        "status": "onboarding",
    },
    {
        "id": "EMP002",
        "name": "Giulia Bianchi",
        "email": "giulia.bianchi@company.com",
        "role": "Product Manager",
        "department": "Product",
        "start_date": "2024-03-01",
        "manager_id": "EMP005",
        "status": "active",
    },
    {
        "id": "EMP003",
        "name": "Alessandro Romano",
        "email": "alessandro.romano@company.com",
        "role": "Engineering Manager",
        "department": "Engineering",
        "start_date": "2023-01-15",
        "manager_id": "EMP005",
        "status": "active",
    },
    {
        "id": "EMP004",
        "name": "Carlos Alcaraz",
        "email": "carlos.alcaraz@company.com",
        "role": "UX Designer",
        "department": "Design",
        "start_date": "2024-06-01",
        "manager_id": "EMP002",
        "status": "active",
    },
    {
        "id": "EMP005",
        "name": "Luca Ferrari",
        "email": "luca.ferrari@company.com",
        "role": "CTO",
        "department": "Executive",
        "start_date": "2022-01-01",
        "manager_id": None,
        "status": "active",
    },
    {
        "id": "EMP006",
        "name": "Roberto Stagi",
        "email": "roberto.stagi@company.com",
        "role": "HR Manager",
        "department": "HR",
        "start_date": "2023-06-01",
        "manager_id": "EMP005",
        "status": "active",
    },
    {
        "id": "EMP007",
        "name": "Jannik Sinner",
        "email": "jannik.sinner@company.com",
        "role": "Sales Rep",
        "department": "Sales",
        "start_date": "2024-01-15",
        "manager_id": "EMP008",
        "status": "active",
    },
    {
        "id": "EMP008",
        "name": "Francesca Gallo",
        "email": "francesca.gallo@company.com",
        "role": "Sales Director",
        "department": "Sales",
        "start_date": "2023-03-01",
        "manager_id": "EMP005",
        "status": "active",
    },
    {
        "id": "EMP009",
        "name": "Andrea Conti",
        "email": "andrea.conti@company.com",
        "role": "DevOps Engineer",
        "department": "Engineering",
        "start_date": "2024-09-01",
        "manager_id": "EMP003",
        "status": "active",
    },
    {
        "id": "EMP010",
        "name": "Laura Russo",
        "email": "laura.russo@company.com",
        "role": "Accountant",
        "department": "Finance",
        "start_date": "2024-04-01",
        "manager_id": "EMP005",
        "status": "active",
    },
]


def get_employee_by_id(employee_id: str) -> Employee | None:
    return next((e for e in employees if e["id"] == employee_id), None)


def get_employees_by_department(department: str) -> list[Employee]:
    return [e for e in employees if e["department"] == department]


def get_employees_by_status(status: str) -> list[Employee]:
    return [e for e in employees if e["status"] == status]


def add_employee(
    *,
    name: str,
    email: str,
    role: str,
    department: str,
    start_date: str,
    manager_id: str | None = None,
) -> Employee:
    new_id = f"EMP{len(employees) + 1:03d}"
    emp: Employee = {
        "id": new_id,
        "name": name,
        "email": email,
        "role": role,
        "department": department,
        "start_date": start_date,
        "manager_id": manager_id,
        "status": "onboarding",
    }
    employees.append(emp)
    return emp


def update_employee(employee_id: str, **updates: str) -> Employee | None:
    emp = get_employee_by_id(employee_id)
    if emp is None:
        return None
    emp.update(updates)  # type: ignore[typeddict-item]
    return emp
